src/main/java/teammates/ui/controller/InstructorFeedbackEditPageData.java

```diff
@@ -113,34 +113,101 @@ private void buildExistingQuestionForm(String feedbackSessionName,
             isResponsesVisibleFor.put(participantType.name(), true);
         }
-        FeedbackQuestionVisibilitySettings visibilitySettings = new FeedbackQuestionVisibilitySettings(
-                                                                        question.getVisibilityMessage(),
-                                                                        isResponsesVisibleFor,
-                                                                        isGiverNameVisibleFor,
-                                                                        isRecipientNameVisibleFor);
-        qnForm.setVisibilitySettings(visibilitySettings);
-        
-        qnForm.setQuestionHasResponses(questionHasResponses.get(question.getId()));
-        
-        qnForm.setQuestionSpecificEditFormHtml(questionDetails.getQuestionSpecificEditFormHtml(questionIndex));
-        qnForm.setEditable(false);
-        
-        qnForms.add(qnForm);
+
+        String dropdownMenuLabel = getDropdownMenuLabel(question);
+
+        return new FeedbackQuestionVisibilitySettings(question.getVisibilityMessage(), isResponsesVisibleFor,
+                                                      isGiverNameVisibleFor, isRecipientNameVisibleFor, dropdownMenuLabel);
+    }
+
+    private String getDropdownMenuLabel(FeedbackQuestionAttributes question) {
+        if (isVisibilitySetToAnonymousToRecipientAndInstructors(question)) {
+            return Const.FeedbackQuestion.COMMON_VISIBILITY_OPTIONS.get("ANONYMOUS_TO_RECIPIENT_AND_INSTRUCTORS");
+        }
+
+        if (isVisibilitySetToAnonymousToRecipientVisibleToInstructors(question)) {
+            return Const.FeedbackQuestion.COMMON_VISIBILITY_OPTIONS.get("ANONYMOUS_TO_RECIPIENT_VISIBLE_TO_INSTRUCTORS");
+        }
+
+        if (isVisibilitySetToVisibleToInstructorsOnly(question)) {
+            return Const.FeedbackQuestion.COMMON_VISIBILITY_OPTIONS.get("VISIBLE_TO_INSTRUCTORS_ONLY");
+        }
+
+        if (isVisibilitySetToVisibleToRecipientAndInstructors(question)) {
+            return Const.FeedbackQuestion.COMMON_VISIBILITY_OPTIONS.get("VISIBLE_TO_RECIPIENT_AND_INSTRUCTORS");
+        }
+
+        return "Custom visibility option:";
+    }
+
+    private boolean isVisibilitySetToAnonymousToRecipientAndInstructors(FeedbackQuestionAttributes question) {
+        boolean responsesVisibleToNotJustGiver = !question.showResponsesTo.isEmpty();
+        boolean giverNameVisibleOnlyToGiver = question.showGiverNameTo.isEmpty();
+
+        return responsesVisibleToNotJustGiver && giverNameVisibleOnlyToGiver;
+    }
+
+    private boolean isVisibilitySetToAnonymousToRecipientVisibleToInstructors(FeedbackQuestionAttributes question) {
+        boolean responsesVisibleToNotJustInstructors = question.showResponsesTo.size() > 1;
+        boolean giverNameVisibleOnlyToInstructors = question.showGiverNameTo.size() == 1
+                && question.showGiverNameTo.contains(FeedbackParticipantType.INSTRUCTORS);
```

Comment:
Wrong logic because over-inclusive: covers anonymous to anyone

Should be `responsesVisibleOnyToRecipientAndInstructors && giverNameVisibleOnlyToInstructors`